independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -323,9 +323,15 @@ private static ScopeInfo initStereotypeScope(List<StereotypeInfo> stereotypes, A
         if (stereotypes.isEmpty()) {
             return null;
         }
-        final Set<ScopeInfo> stereotypeScopes = new HashSet<>();
-        for (StereotypeInfo stereotype : stereotypes) {
-            stereotypeScopes.add(stereotype.getDefaultScope());
+        // first go through all standard stereotypes
+        Set<ScopeInfo> stereotypeScopes = stereotypes.stream().filter(s -> !s.isAdditionalBeanDefiningAnnotation())
```

Comment:
What's wrong with adding one `if` statement to the original for loop? I mean the stream and two more lambdas don't bring any benefit here, except that it's harder to read and few more objects are created ;-).

Reply:
@mkouba better? :)